megamek/src/megamek/common/Crew.java

```diff
@@ -1354,10 +1385,64 @@ public boolean doConsoleRoleSwap() {
         return false;
     }
 
+    //region extraData
+    public void initializeExtraDataAsHashMap(int crewNumber) {
+        this.extraData = new HashMap<>(crewNumber);
+        for (int i = 0; i < crewNumber; i++) {
+            this.extraData.put(i, new HashMap<>());
```

Comment:
What is the expected size of these hash maps?

Reply:
Maximum would be crew size, so in most cases (given the primary use case is for OpFor generation in MekHQ) under 10 instances for the first, then 2 to 4 values for the second set of maps.

Reply:
My preference would be to not initialize the crew member slots to anything.